packages/core/src/transport/startBatchWithReplica.ts

```diff
@@ -3,24 +3,36 @@ import type { Context } from '../tools/serialisation/context'
 import type { Observable } from '../tools/observable'
 import type { PageExitEvent } from '../browser/pageExitObservable'
 import type { RawError } from '../domain/error/error.types'
+import type { Encoder } from '../tools/encoder'
+import { createIdentityEncoder } from '../tools/encoder'
 import { Batch } from './batch'
 import { createHttpRequest } from './httpRequest'
 import { createFlushController } from './flushController'
 
+interface BatchConfiguration {
+  endpoint: EndpointBuilder
+  encoder?: Encoder
+}
+
+interface ReplicaBatchConfiguration<T> extends BatchConfiguration {
+  transformMessage?: (message: T) => T
+}
+
 export function startBatchWithReplica<T extends Context>(
   configuration: Configuration,
-  primary: { endpoint: EndpointBuilder },
-  replica: { endpoint: EndpointBuilder; transformMessage?: (message: T) => T } | undefined,
+  primary: BatchConfiguration,
+  replica: ReplicaBatchConfiguration<T> | undefined,
   reportError: (error: RawError) => void,
   pageExitObservable: Observable<PageExitEvent>,
   sessionExpireObservable: Observable<void>
 ) {
-  const primaryBatch = createBatch(configuration, primary.endpoint)
-  const replicaBatch = replica && createBatch(configuration, replica.endpoint)
+  const primaryBatch = createBatch(configuration, primary)
+  const replicaBatch = replica && createBatch(configuration, replica)
 
-  function createBatch(configuration: Configuration, endpointBuilder: EndpointBuilder) {
+  function createBatch(configuration: Configuration, { endpoint, encoder }: BatchConfiguration) {
     return new Batch(
-      createHttpRequest(configuration, endpointBuilder, configuration.batchBytesLimit, reportError),
+      encoder || createIdentityEncoder(),
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I had a hard time understanding in which case the encoder was undefined and I'm wondering if it could me simplified. FMU we have to look at: 
- Identify that createDeflateEncoder can be undefined : https://github.com/DataDog/browser-sdk/blob/9862d64c4995ecf805f2dac4f357e3bcda387ed6/packages/rum-core/src/transport/startRumBatch.ts#L36
- Because  deflateWorker && createDeflateEncoder can be undefined : https://github.com/DataDog/browser-sdk/blob/9862d64c4995ecf805f2dac4f357e3bcda387ed6/packages/rum-core/src/boot/rumPublicApi.ts#L193-L195
- Then deflateWorker  can be null because of: https://github.com/DataDog/browser-sdk/blob/9862d64c4995ecf805f2dac4f357e3bcda387ed6/packages/rum-core/src/boot/rumPublicApi.ts#L152-L161

Personally I prefer to have a `rumPublicApi` as dump as possible and make the instantiation checks in startRum.
Also, It could be nice to create the encoder in startRum() to remove the logic from `startRumBatch` and `startBatchWithReplica`.  
I don't know all the intricacies of the implement so it might no be feasible 🙂 Let me know what you think.

Reply:
As seen IRL, I changed a bit how encoders are created so the condition on whether we use an "Identity" or "Deflate" encoder is clearer.